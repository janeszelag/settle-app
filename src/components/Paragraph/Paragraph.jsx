import React from "react"
import styled from "styled-components/native"
import PropTypes from "prop-types"

const fontSizes = {
  small: 14,
  medium: 16,
  large: 18
}

const paragraphSize = ({ size }) => ({ fontSize: fontSizes[size] })
const paragraphAlign = ({ align }) => ({ textAlign: align })
const StyledText = styled.Text(
  paragraphSize,
  paragraphAlign,
  ({ bold }) => {
    return {
      fontWeight: bold ? 'bold' : 'normal'
    }
  }
)

const Paragraph = ({ bold, align, size, children }) => {


  return (
    <StyledText size={size} bold={bold} align={align}>
      {children}
    </StyledText>
  )
}

Paragraph.propTypes = {
  /**
   * Embolden paragraph text without conveying any special importance or relevance.
   */
  bold: PropTypes.bool,
  /**
   * The font size.
   */
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  /**
   * Content alignment.
   */
  align: PropTypes.oneOf(['left', 'center', 'right']),
  /**
   * The content. Can be text, other components.
   */
  children: PropTypes.node.isRequired,
}

Paragraph.defaultProps = {
  bold: false,
  size: 'medium',
  align: 'left',
}

export default Paragraph