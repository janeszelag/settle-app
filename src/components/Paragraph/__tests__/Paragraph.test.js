import React from "react"
import renderer from "react-test-renderer"
import Paragraph from "../Paragraph"

describe("<Paragraph />", () => {
  it("has 1 child", () => {
    const tree = renderer
      .create(<Paragraph>Paragraph</Paragraph>)
      .toJSON()
    expect(tree.children.length).toBe(1)
  })

  it("renders at medium size", () => {
    const tree = renderer
      .create(<Paragraph>Paragraph</Paragraph>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders medium and bold", () => {
    const tree = renderer
      .create(<Paragraph bold={true}>Paragraph</Paragraph>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders at small size", () => {
    const tree = renderer
      .create(<Paragraph small='small'>Paragraph</Paragraph>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders at large size", () => {
    const tree = renderer
      .create(<Paragraph size='large'>Paragraph</Paragraph>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
