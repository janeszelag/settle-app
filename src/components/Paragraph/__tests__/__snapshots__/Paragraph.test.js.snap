// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> renders at large size 1`] = `
<Text
  align="left"
  bold={false}
  size="large"
  style={
    Array [
      Object {
        "fontSize": 18,
        "fontWeight": "normal",
        "textAlign": "left",
      },
    ]
  }
>
  Paragraph
</Text>
`;

exports[`<Paragraph /> renders at medium size 1`] = `
<Text
  align="left"
  bold={false}
  size="medium"
  style={
    Array [
      Object {
        "fontSize": 16,
        "fontWeight": "normal",
        "textAlign": "left",
      },
    ]
  }
>
  Paragraph
</Text>
`;

exports[`<Paragraph /> renders at small size 1`] = `
<Text
  align="left"
  bold={false}
  size="medium"
  style={
    Array [
      Object {
        "fontSize": 16,
        "fontWeight": "normal",
        "textAlign": "left",
      },
    ]
  }
>
  Paragraph
</Text>
`;

exports[`<Paragraph /> renders medium and bold 1`] = `
<Text
  align="left"
  bold={true}
  size="medium"
  style={
    Array [
      Object {
        "fontSize": 16,
        "fontWeight": "bold",
        "textAlign": "left",
      },
    ]
  }
>
  Paragraph
</Text>
`;
